core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -300,23 +299,12 @@ public DeleteWriteBuilder equalityFieldIds(int... fieldIds) {
 
       meta("delete-type", "position");
 
-      if (rowSchema != null && createWriterFunc != null) {
-        // the appender uses the row schema wrapped with position fields
-        appenderBuilder.schema(new org.apache.iceberg.Schema(
-            MetadataColumns.DELETE_FILE_PATH,
-            MetadataColumns.DELETE_FILE_POS,
-            NestedField.optional(
-                MetadataColumns.DELETE_FILE_ROW_FIELD_ID, "row", rowSchema.asStruct(),
-                MetadataColumns.DELETE_FILE_ROW_DOC)));
+      appenderBuilder.schema(DeletesUtil.posDeleteSchema(rowSchema));
```

Comment:
This changes the logic so that it is incorrect. If a row will not be written, then the schema should not include that row. The row may not be written if either the `rowSchema` is not set, or if the `createWriterFunc` is not set. But there are cases where the `rowSchema` is set and the other is not because of the way callers use this API.

Reply:
Agree,  the current version is incorrect. I will revert it. Thanks.